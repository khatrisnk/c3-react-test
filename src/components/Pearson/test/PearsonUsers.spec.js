import React from 'react';
import PearsonUsersComponent from '../PearsonUsers';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

const usersList = [
  {
    id: 4,
    first_name: 'Eve',
    last_name: 'Holt',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg'
  },
  {
    id: 5,
    first_name: 'Charles',
    last_name: 'Morris',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg'
  },
  {
    id: 6,
    first_name: 'Tracey',
    last_name: 'Ramos',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg'
  }
];

const usersListAfterDelete = [
  {
    id: 4,
    first_name: 'Eve',
    last_name: 'Holt',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg'
  },
  {
    id: 5,
    first_name: 'Charles',
    last_name: 'Morris',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg'
  }
];

const state = {
  userToAdd: [
    {
      id: 4,
      first_name: 'Eve',
      last_name: 'Holt',
      avatar:
        'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg'
    }
  ]
};

describe('PearsonUsersComponent Sanpshots', () => {
  it('PearsonUsersComponent Sanpshot - with Loading... text', () => {
    const component = renderer.create(
      <PearsonUsersComponent pearsonUsers={[]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('PearsonUsersComponent Sanpshot - with users list', () => {
    const component = renderer.create(
      <PearsonUsersComponent pearsonUsers={usersList} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('PearsonUsersComponent', () => {
  it('renders three <User /> components', () => {
    const wrapper = render(<PearsonUsersComponent pearsonUsers={usersList} />);
    expect(wrapper.find('.user')).toHaveLength(3);
  });
  it('renders one heading', () => {
    const wrapper = shallow(<PearsonUsersComponent pearsonUsers={usersList} />);
    const heading = (
      <h1 className="user-mgmt-heading">Pearson User Management</h1>
    );
    expect(wrapper.contains(heading)).toEqual(true);
  });
  it('renders Loading... when user list is empty', () => {
    const wrapper = shallow(<PearsonUsersComponent pearsonUsers={[]} />);
    expect(wrapper.find('.user-loader')).toHaveLength(1);
  });
  it('user method returns unique users', () => {
    const wrapper = mount(<PearsonUsersComponent pearsonUsers={usersList} />);
    expect(
      PearsonUsersComponent.getDerivedStateFromProps(wrapper.props(), state)
        .users
    ).toEqual(usersList);
  });
  it('deleteuser method delete one user from the userlist', () => {
    const e = {
      preventDefault: jest.fn()
    };
    const id = 6;
    const wrapper = mount(
      <PearsonUsersComponent
        pearsonUsers={usersList}
        deletePearsonUser={jest.fn()}
      />
    );
    expect(wrapper.instance().deleteUser(id, e)).toEqual(usersListAfterDelete);
  });
});
