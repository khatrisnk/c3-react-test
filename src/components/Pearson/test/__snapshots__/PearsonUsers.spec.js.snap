// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PearsonUsersComponent Sanpshots PearsonUsersComponent Sanpshot - with Loading... text 1`] = `
<div
  className="user-loader"
>
  Loading...
</div>
`;

exports[`PearsonUsersComponent Sanpshots PearsonUsersComponent Sanpshot - with users list 1`] = `
<div
  className="pearson-users"
>
  <div
    className="container row"
  >
    <h1
      className="user-mgmt-heading"
    >
      Pearson User Management
    </h1>
  </div>
  <div
    className="user-container container row"
  >
    <div
      className="user col-md-4"
    >
      <div
        className="user-top"
      />
      <div
        className="user-bottom"
      >
        <div
          className="user-name"
        >
          <span
            className="user-first-name"
          >
            Eve
          </span>
           
          <span
            className="user-last-name"
          >
            Holt
          </span>
        </div>
        <a
          className="user-delete"
          href="# "
          onClick={[Function]}
          title="delete"
        >
          Delete
        </a>
      </div>
      <img
        alt="avtar"
        className="user-avatar"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
      />
    </div>
    <div
      className="user col-md-4"
    >
      <div
        className="user-top"
      />
      <div
        className="user-bottom"
      >
        <div
          className="user-name"
        >
          <span
            className="user-first-name"
          >
            Charles
          </span>
           
          <span
            className="user-last-name"
          >
            Morris
          </span>
        </div>
        <a
          className="user-delete"
          href="# "
          onClick={[Function]}
          title="delete"
        >
          Delete
        </a>
      </div>
      <img
        alt="avtar"
        className="user-avatar"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
      />
    </div>
    <div
      className="user col-md-4"
    >
      <div
        className="user-top"
      />
      <div
        className="user-bottom"
      >
        <div
          className="user-name"
        >
          <span
            className="user-first-name"
          >
            Tracey
          </span>
           
          <span
            className="user-last-name"
          >
            Ramos
          </span>
        </div>
        <a
          className="user-delete"
          href="# "
          onClick={[Function]}
          title="delete"
        >
          Delete
        </a>
      </div>
      <img
        alt="avtar"
        className="user-avatar"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
      />
    </div>
  </div>
</div>
`;
